crates/ruff_python_formatter/src/expression/mod.rs

```diff
@@ -1007,6 +1025,75 @@ pub(crate) fn has_own_parentheses(
     }
 }
 
+/// Returns `true` if the expression can hug directly to enclosing parentheses.
+///
+/// For example, in preview style, given:
```

Comment:
Should we move the preview check into `is_huggable` to avoid the repetition?

Reply:
Will do.